tests/fixtures/cwd_project/pyproject.toml

```diff
@@ -1,3 +1,11 @@
 [tool.poe.tasks.cwd]
 cmd = "poe_test_pwd"
 cwd = "./subdir/foo"
+
+[tool.poe.tasks.cwd_env]
+cmd = "poe_test_pwd"
+cwd = "./subdir/${BAR_ENV}"
```

Comment:
Good catch, it helped find a bug, as I was using `self.env` in `get_executor` instead of the `env` parameter. For reasons I did not care too much to track down `self.env` does not have the variables generated from program arguments, but the `env` parameter does.

Anyway, I replaced `self.env` with `env` and it seems to be working correctly now.